src/PixelRendr.ts

```diff
@@ -257,16 +254,34 @@ export class PixelRendr implements IPixelRendr {
      * @param key   The key of the sprite to render.
      */
     public resetRender(key: string): void {
-        const render: IRender = this.BaseFiler.get(key);
+        const result: IRender | IRenderLibrary = this.BaseFiler.get(key);
 
-        if (!render || !render.sprites) {
+        if (result === this.library.sprites) {
             throw new Error(`No render found for '${key}'.`);
         }
 
+        const render: IRender = result as IRender;
+
         render.sprites = {};
     }
 
     /**
+     * Replaces the current palette with a new one.
+     * 
+     * @param palette   The new palette to replace the current one.
+     */
+    public changePalette(palette: IPalette): void {
+        this.setPalette(palette);
+
+        for (let sprite in this.library.sprites) {
```

Comment:
Nit: can you use `const` instead of `let` here? I went through a bunch of older projects and fixed that up.